### PR TITLE
yargs: Add 'sort-commands' option to `parserConfiguration`

### PR DESCRIPTION
**Pull request by cameronhunter**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:
 
- http://yargs.js.org/docs/#api-parserconfigurationobj
- https://github.com/yargs/yargs/blob/05324b426025f7bc52b0caeb0794dd61abdd6845/docs/api.md#parserconfigurationobj
- https://github.com/yargs/yargs/blob/63b3dd31a455d428902220c1992ae930e18aff5c/lib/usage.js#L209

Although it looks as if this belongs in `yargs-parser`, it's handled directly in `yargs` (see last code link)

### CODE REVIEW

File: types/yargs/index.d.ts [hunk 1]
```diff
@@ -427,7 +427,10 @@ declare namespace yargs {
         parsed: DetailedArguments | false;
 
         /** Allows to configure advanced yargs features. */
-        parserConfiguration(configuration: Partial<Configuration>): Argv<T>;
+        parserConfiguration(configuration: Partial<Configuration & {
+            /** Sort commands alphabetically. Default is `false` */
+            'sort-commands': boolean;
```
Comment (pushplay) [inline types/yargs/index.d.ts]: Intersection type in the generic type param is kind of gross.  Can we pull it out and name it to clean this up?

**Review by pushplay — CHANGES_REQUESTED** [review]
I think the intersection type in the generic type param should be cleaned up.

File: types/yargs/index.d.ts (hunk 1 above)
Comment (cameronhunter) [reply types/yargs/index.d.ts]: Sure thing. I've pulled it out to a defined type instead of being inline. Thanks for the feedback.

**Review by pushplay — APPROVED** [review]
Great.  Thanks for the clean up.

**Pull request merged**